### DISCUSSION TITLE
Support Custom Roles/Claims with OIDC

### DISCUSSION THREAD

**Discussion by ncecere** [open-webui/open-webui#6152] [General]
Adding the ability to define and pass in custom claims for auth would enable AuthZ support for the application.  Login and access could be scoped to a subset of users instead of an entire organization.  This could also be used with the #2924 request since user could be assigned to specific groups at login instead of being assigned by administrators.

**Comment by hubby2004** [comment]
This will be a great feature if we can restrict access based on a specific attribute and its value. For instance, we can release an eduPersonEntitlement attribute as part of the OIDC flow and have the open webui allow only those whose eduPersonEntitlement value contains a specific string. EduPersonEntitlement is a multi value attribute so supporting a list solves the issue of single and multi value attributes.

**Comment by ncecere** [comment]
I think it would need to be added around [here](https://github.com/open-webui/open-webui/blob/e8babe62bc8e466be0367703fd062a981f5c2394/backend/open_webui/main.py#L2262) but I'm not 100% sure. And think this would also solve the feature in  #4858 that @atnjqt opened.
